webui/src/pages/user/user-namespace-extension-list.tsx

```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ * Copyright (c) 2020 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { Theme, createStyles, WithStyles, withStyles, Typography } from '@material-ui/core';
+import { Namespace, isError, Extension, ErrorResult } from '../../extension-registry-types';
+import { UserNamespaceExtensionListItem } from './user-namespace-extension-list-item';
+import { ExtensionRegistryService } from '../../extension-registry-service';
+import { ErrorResponse } from '../../server-request';
+import { DelayedLoadIndicator } from '../../custom-mui-components/delayed-load-indicator';
+
+const extensionListStyles = (theme: Theme) => createStyles({
+    container: {
+        padding: theme.spacing(5, 0),
+    },
+    extensions: {
+        display: 'grid',
+        gridTemplateColumns: `repeat(auto-fit, minmax(200px, 1fr))`,
+        gap: `1.5rem`,
+        marginTop: '1rem',
+    }
+});
+
+
+class UserNamespaceExtensionListComponent extends React.Component<UserNamespaceExtensionListComponent.Props, UserNamespaceExtensionListComponent.State> {
+
+    constructor(props: UserNamespaceExtensionListComponent.Props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            extensions: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.initExtensions();
+    }
+
+    async initExtensions() {
+        const extensionsURLs: string[] = Object.keys(this.props.namespace.extensions).map((key: string) => this.props.namespace.extensions[key]);
+
+        const getExtension = async (url: string) => {
+            let result: Extension | ErrorResult;
+            try {
+                result = await this.props.service.getExtensionDetail(url);
+                if (isError(result)) {
+                    throw result;
+                }
+                return result;
+            } catch (error) {
+                this.props.setError(error);
+                return undefined;
+            }
+        };
+
+        const getExtensions = async () => Promise.all(
+            extensionsURLs.map(
+                (url: string) => getExtension(url)
+            )
+        );
+
+        const extensions = await getExtensions();
+        if (extensions.length) {
```

Comment:
The namespace could be empty (have no extensions), so it's ok to set the state even if the `extensions` array is empty. Otherwise the load indicator would stay active all the time.